### ISSUE TITLE
Issue with Array.prototype.filter

### ISSUE THREAD

**Issue by Promo656** [greatfrontend/greatfrontend#26]
## Your GreatFrontEnd account email (optional)

rank2007@mail.ru 

## Problematic section

- [ ] Question Description
- [ ] Question Skeleton
- [x] Question Solution
- [ ] Test Cases (Missing/Incorrect/Unclear)

## Type of question

- [x] JavaScript
- [ ] User Interface
- [ ] System Design
- [ ] Quiz

## Describe the issue

Just replace `this[k]` with `kValue` in the example code below

## Code you used to submit (where relevant)

```js
Array.prototype.myFilter = function (callbackFn, thisArg) {
  if (
    typeof callbackFn !== 'function' ||
    !callbackFn.call ||
    !callbackFn.apply
  ) {
    throw new TypeError(`${callbackFn} is not a function`);
  }

  const len = this.length;
  const array = [];
  let k = 0;
  let to = 0;

  while (k < len) {
    // Ignore index if value is not defined for index (e.g. in sparse arrays).
    const kPresent = Object.hasOwn(this, k);
    if (kPresent) {
      const kValue = this[k];
      const selected = Boolean(callbackFn.call(thisArg, kValue, k, this));
      if (selected) {
        array[to] = this[k]; // <----------------------------------------------------------- replace
        to++;
      }
    }
    k++;
  }

  return array;
};
```

**Comment by gfeeng** [comment]
Have fixed this, thank you!

**Comment by Promo656** [comment]
﻿﻿What about reward?On Jan 24, 2023, at 09:01, GreatFrontEnd ***@***.***> wrote:﻿
Closed #26 as completed.
